compress/zstd/zstd.go

```diff
@@ -66,5 +66,8 @@ func (z *ZSTD) Compress(data []byte) ([]byte, error) {
 		return data, err
 	}
 
-	return tmpBuffer.Bytes(), nil
```

Comment:
Could you explain more how it was creating problems ?

Is it because `tmpBuffer.Bytes()` is a pointer to the same slice across call to Compress ?

Reply:
Yeah, exactly. In Go, a slice is a pointer + capacity + length. In this case, they all shared the same pointer, but a different length.

Reply:
ah, that could explain the weird issue with truncated data !